astronomer/providers/amazon/aws/triggers/redshift_cluster.py

```diff
@@ -46,26 +45,28 @@ async def run(self):
         if operation_type is 'pause_cluster it will call the pause_cluster function in RedshiftHookAsync
         """
         hook = RedshiftHookAsync(aws_conn_id=self.aws_conn_id)
-        while True:
-            try:
-                if self.operation_type == "resume_cluster":
-                    response = await hook.resume_cluster(cluster_identifier=self.cluster_identifier)
-                    if response:
-                        yield TriggerEvent(response)
-                        return
-                    else:
-                        error_message = f"{self.task_id} failed"
-                        raise AirflowException(error_message)
+        try:
+            if self.operation_type == "resume_cluster":
+                response = await hook.resume_cluster(cluster_identifier=self.cluster_identifier)
+                if response:
+                    yield TriggerEvent(response)
+                    return
                 else:
-                    response = await hook.pause_cluster(cluster_identifier=self.cluster_identifier)
-                    if response:
-                        yield TriggerEvent(response)
-                        return
-                    else:
-                        error_message = f"{self.task_id} failed"
-                        raise AirflowException(error_message)
-            except AirflowException:
-                await asyncio.sleep(self.polling_period_seconds)
+                    error_message = f"{self.task_id} failed"
+                    yield TriggerEvent({"status": "error", "message": str(error_message)})
+                    return
+            else:
+                response = await hook.pause_cluster(cluster_identifier=self.cluster_identifier)
+                if response:
+                    yield TriggerEvent(response)
+                    return
+                else:
+                    error_message = f"{self.task_id} failed"
+                    yield TriggerEvent({"status": "error", "message": str(error_message)})
+                    return
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e)})
+            return
```

Comment:
I thought only tests are supposed to be fixed.Why are code changes being done to address code coverage. Did we miss any code during the merge ?